### PR TITLE
ci: fix packages build for ee releases 

### PR DESCRIPTION
**Pull request by id**
- build otp manually instead of kerl
- bump versions on some actions
- fix deprecated way of setting step output in some cases

### CODE REVIEW

**Review by thalesmg — APPROVED** [review]

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -59,12 +54,10 @@ jobs:
...
                -d "{\"repo\":\"emqx/emqx\", \"tag\": \"${{ github.ref_name }}\" }" \
                ${{ secrets.EMQX_IO_RELEASE_API }}
-      - uses: actions/checkout@v2
-        with:
-          fetch-depth: 0
+      - uses: actions/checkout@v3
       - name: get version
         id: version
-        run: echo "::set-output name=version::$(./pkg-vsn.sh)"
+        run: echo "{version}={$(./pkg-vsn.sh)}" >> $GITHUB_OUTPUT
```
Comment (thalesmg) [inline .github/workflows/release.yaml:60]:
```suggestion
        run: echo "version=$(./pkg-vsn.sh)" >> $GITHUB_OUTPUT
```

**Review by thalesmg — DISMISSED** [review]

**Review by JimMoen — DISMISSED** [review]

File: .github/actions/detect-profiles/action.yaml [hunk 2]
```diff
@@ -15,11 +11,9 @@ runs:
       shell: bash
       run: |
         if make emqx-ee --dry-run > /dev/null 2>&1; then
-          echo "::set-output name=profiles::[\"emqx-ee\"]"
-          echo "https://ci%40emqx.io:${{ inputs.ci_git_token }}@github.com" > $HOME/.git-credentials
-          git config --global credential.helper store
+          echo "{profiles}={[\"emqx-ee\"]}" >> $GITHUB_OUTPUT
```
[the remark this answers is not in the document]
Comment (id) [reply .github/actions/detect-profiles/action.yaml]: I hid my embarrassment by amending the commit

**Comment by zmstone** [comment]
nit: commit message should be `ci: ...` not `chore:`

**Review by thalesmg — APPROVED** [review]

**Comment by id** [comment]
@thalesmg sorry for dismissing your review, again

**Review by thalesmg — APPROVED** [review]

**Comment by id** [comment]
Something borked the CI, created another PR https://github.com/emqx/emqx/pull/9270

**Pull request closed without merging**